apps/verceldashboard/vercel_dashboard.star

```diff
@@ -0,0 +1,168 @@
+"""
+Applet: Vercel Dashboard
+Summary: Monitor Vercel information
+Description: Monitor your Vercel deployments and view important information.
+Author: Chase Roossin
+"""
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+BASE_DEPLOYMENT_URL = "https://api.vercel.com/v6/deployments"
+CACHE_KEY = "vercel-cache"
+SPACE = "   "
+
+# Config
+CONFIG_API_KEY = "api-key"
+CONFIG_CLOCK_FORMAT = "time-format"
+CONFIG_TIMEZONE = "time-zone"
+
+def main(config):
+    apikey = config.str(CONFIG_API_KEY)
+    showIn24hr = config.bool(CONFIG_CLOCK_FORMAT, True)
+
+    # Figure out how to show if no api key
+    if apikey == None:
+        fail("No API Key provided")
+
+    cached_data = cache.get(CACHE_KEY)
```

Comment:
Cache is scoped to your app, but not to individual users. This record will hold whatever happened to get written to cache last time. You likely need to incorporate something like the API key into the cache key.

Reply:
Oh very interesting, I did not realize that when reading the docs - must have overlooked. Will fix that.

Reply:
@matslina Do we need to encode/decode the API key if we are going to use that as part of the cache key? Or is the security in place that the cache key is not exposed anywhere?

Reply:
The cache key will not be visible to other apps, so using it directly should be secure. That said, there's nothing wrong with computing a hash of it and using that instead.

Reply:
Makes sense - I have went ahead and updated it with a hashed version of the api key!